### PR TITLE
Implemented CheckBanned and Unban (Attempt two)

### PR DESCRIPTION
**Pull request by jonfriesen**
Built two additional commands (unban, checkbanned) and made some changes to
the existing ban command.

**Changes**

ban now shows notification to admin user in bad
custom message if admin tries to ban themselves

**Additions**

unban command: _/unban username_
checkbanned command: _/checkbanned_ or _/checkbanned user_ (shows all banned users or
if a specific users banned status)
I built this PR based on #1075 and it should be noted that the existing
ban command already works on offline users.

I've also added unit tests for both unban and checkbanned commands.

This is my second attempt at posting this. The first time went horrible because I can't GIT. This time I've used an up-to-date branch so my PR shouldn't be stale. Also all commits are squashed instead of a ton pushing.

Hope this is up to your standards, if I can do anything better regarding code style, git pushing, etc I would greatly appreciate the advice!

### CODE REVIEW

**Comment by hnrkndrssn** [comment]
Looks good to me

File: JabbR/Hubs/Chat.cs [hunk 1]
```diff
@@ -1199,6 +1199,38 @@ void INotificationService.BanUser(ChatUser targetUser)
...
+        {
+            Clients.Caller.unbanUser(new
+            {
+                Name = targetUser.Name
+            });
+        }
+
+        void INotificationService.IsBanned()
+        {
+            // Get all users that are banned
```
Comment (samandmoore) [inline JabbR/Hubs/Chat.cs:1219]: is this comment really useful?

**Comment by samandmoore** [comment]
this looks great. it would be swell to also have some tests added for it though: like this. https://github.com/JabbR/JabbR/blob/dev/JabbR.Tests/CommandManagerFacts.cs#L2365

**Comment by jonfriesen** [comment]
@samandmoore, thanks for the advice! Added unit tests for both commands. I'm still grasping how mocking works, so (I think) I'm a little light in that area still.

**Comment by samandmoore** [comment]
lgtm with the tests.

File: JabbR/Services/INotificationService.cs [hunk 2]
```diff
@@ -39,6 +39,9 @@ public interface INotificationService
         void AllowUser(ChatUser targetUser, ChatRoom targetRoom);
         void UnallowUser(ChatUser targetUser, ChatRoom targetRoom);
         void BanUser(ChatUser targetUser);
+        void UnbanUser(ChatUser targetUser);
+        void IsBanned(ChatUser targetUser);
+        void IsBanned();
```
Comment (davidfowl) [inline JabbR/Services/INotificationService.cs]: This name needs to change across the board. This should be CheckBanned

**Pull request merged**